azurerm/resource_arm_private_link_service_test.go

```diff
@@ -0,0 +1,303 @@
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_private_link_service" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		if resp, err := client.Get(ctx, resourceGroup, name, ""); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Bad: Get on network.PrivateLinkServiceClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMPrivateLinkService_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_private_link_service" "test" {
+  name                           = "acctestpls-%d"
+  location                       = azurerm_resource_group.test.location
+  resource_group_name            = azurerm_resource_group.test.name
+
+  primary_nat_ip_configuration {
+    name                         = "primaryIpConfiguration-%d"
+    subnet_id                    = azurerm_subnet.test.id
+  }
+
+  load_balancer_frontend_ip_configuration_ids = [
+    azurerm_lb.test.frontend_ip_configuration.0.id
+  ]
+}
+`, testAccAzureRMPrivateLinkServiceTemplate(rInt, location), rInt, rInt)
+}
+
+func testAccAzureRMPrivateLinkService_update(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_private_link_service" "test" {
+  name                           = "acctestpls-%d"
+  location                       = azurerm_resource_group.test.location
+  resource_group_name            = azurerm_resource_group.test.name
+
+  primary_nat_ip_configuration {
+    name                         = "primaryIpConfiguration-%d"
+    subnet_id                    = azurerm_subnet.test.id
+  }
+
+  load_balancer_frontend_ip_configuration_ids = [
+    azurerm_lb.test.frontend_ip_configuration.0.id
+  ]
+
+  tags = {
+    env = "test"
+  }
+}
+`, testAccAzureRMPrivateLinkServiceTemplate(rInt, location), rInt, rInt)
+}
+
+func testAccAzureRMPrivateLinkService_complete(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_private_link_service" "test" {
+  name                           = "acctestpls-%d"
+  location                       = azurerm_resource_group.test.location
+  resource_group_name            = azurerm_resource_group.test.name
+  auto_approval_subscription_ids = [data.azurerm_subscription.current.subscription_id]
+  visibility_subscription_ids    = [data.azurerm_subscription.current.subscription_id]
+
+  primary_nat_ip_configuration {
+    name                         = "primaryIpConfiguration-%d"
+    subnet_id                    = azurerm_subnet.test.id
+    private_ip_address           = "10.5.1.17"
+    private_ip_address_version   = "IPv4"
+  }
+
+  secondary_nat_ip_configuration {
+    name                         = "secondaryIpConfiguration-%d"
+    subnet_id                    = azurerm_subnet.test.id
+    private_ip_address           = "10.5.1.18"
+    private_ip_address_version   = "IPv4"
+  }
```

Comment:
Could we have another test that added/moves these around?

Reply:
Fixed.